app/src/main/java/org/mozilla/fenix/wallpapers/WallpapersUseCases.kt

```diff
@@ -258,6 +258,13 @@ class WallpapersUseCases(
             } else {
                 settings.currentWallpaperName
             }
+            // The card colors should NOT be migrated if the file migration was ran prior to these
+            // changes and it failed. We can verify this by checking [settings.currentWallpaperTextColor],
+            // since this is only initialized for legacy wallpapers in
+            // [LegacyWallpaperMigration.migrateLegacyWallpaper].
+            if (settings.shouldMigrateLegacyWallpaperCardColors && settings.currentWallpaperTextColor != 0L) {
+                migrationHelper.migrateExpiredWallpaperCardColors(currentWallpaperName)
+            }
```

Comment:
Would it be possible to move this logic about the `currentWallpaperTextColor` and the relevant comment into the migration helper itself? I'm glad you found something that worked, but I would prefer to keep the migration hacks we're needing to do encapsulated into that class as much as possible.

Reply:
My suggestion would be:
```
// Usecases:
...
if (settings.shouldMigrateLegacyWallpaperCardColors) {
    migrationHelper.migrateExpiredWallpaperCardColors(currentWallpaperName)
}

//MigrationHelper
fun migrateExpiredWallpaperCardColors(wallpaperName: String) {
     // The card colors should NOT be migrated if the file migration was ran prior to these
     // changes and it failed. We can verify this by checking [settings.currentWallpaperTextColor],
     // since this is only initialized for legacy wallpapers in
    // [LegacyWallpaperMigration.migrateLegacyWallpaper].
    if(settings.currentWallpaperTextColor != 0L) (
...
}
```

It's a bit odd, so it's up to you. But I like that it keeps more of the details of the migration code within the helper itself